src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -249,20 +250,17 @@ export class CellExecution {
     private async completedWithErrors(error: Partial<Error>) {
         traceCellMessage(this.cell, 'Completed with errors');
         this.sendPerceivedCellExecute();
-        if (!this.isEmptyCodeCell) {
-            await chainWithPendingUpdates(this.cell.notebook, (edit) => {
-                traceCellMessage(this.cell, 'Update run run duration');
-                const metadata = this.cell.metadata.with({ lastRunDuration: this.stopWatch.elapsedTime });
-                edit.replaceNotebookCellMetadata(this.cell.notebook.uri, this.cell.index, metadata);
-            });
-        }
-        await updateCellWithErrorStatus(this.cell, error);
 
-        this.errorHandler.handleError((error as unknown) as Error).ignoreErrors();
+        await chainWithPendingUpdates(this.cell.notebook, async () => {
+            traceCellMessage(this.cell, 'Update with error state & output');
+            await this.task?.appendOutput([translateErrorOutput(createErrorOutput(error))]);
```

Comment:
Going to look at the API now, but wouldn't an error be a replaceOutput and not an appendOutput?

Reply:
We don't want to blow away existing output sent by kernel, that doesn't seem right.
We should still display what was returned by the kernel.

FYI - prior to a cell getting executed all existing output is cleared.